packages/program/src/contract.test.ts

```diff
@@ -36,13 +38,49 @@ describe('Contract', () => {
     expect(contract.account).toBe(null);
   });
 
+  test('Contract instance with provider can get balance', async () => {
+    const provider = new Provider('http://localhost:4000/graphql');
+    const contract = new Contract(CONTRACT_ID, ABI, provider);
+    expect(await contract.getBalance([1])).toStrictEqual(bn(0));
```

Comment:
I feel like we can slightly improve these assertions. These assertions are checking if the `getBalance` function returns `0` for a non-existent asset ID IIUC. We should instead be checking that the function returns the _correct_ balance for an actual asset ID with a non-zero balance. Thoughts?

Reply:
Makes sense.

Reply:
Thank you @Dhaiwat10 I will make that change. Currently in a position where I am still getting a feel for the amount of testing required at this code level, a lot of what I am putting through at the moment are validation like tests.

I'm also seeing a lot of places where we are not mocking classes that are not under test within the scope of the unit test. In this test, the `Provider` would be an example of that. I think we should be mocking this so we are not dependent on the behaviour or state of other dependencies for our unit test. I'd be quite keen to refactor this throughout the SDK. What are your thoughts on this?

Reply:
@danielbate I agree. However, adjusting it for the whole repo _(at once)_ can become quite an endless task. Doing it in a few places first might give an idea of how difficult it will be, and then doing it in smaller amounts _(per package or test file)_ would make it easier to review.

Also, since many things are interconnected, I believe we will need to share mock helpers between packages to ease the pain of constructing mocks from scratch for every test. I did similar setups in the past; we can talk about it later if you'd like.

Reply:
@arboleya sounds good and that would be great, thankyou!